opensafely/codelists.py

```diff
@@ -123,8 +123,13 @@ def update(codelists_dir=None):
     preserve_download_dates(manifest, manifest_file)
     manifest_file.write_text(json.dumps(manifest, indent=2))
     for file in old_files - new_files:
-        print(f"Deleting {file.name}")
-        file.unlink()
+        if delete_unknown:
```

Comment:
The "Moderate update's' destructive behaviour" commit is no longer needed for this PR to work, I think? Could you move it to a separate one, so we can think about it separately, please? It's a behaviour change to an existing piece of functionality, so it's probably easiest all round if we avoid letting it interfere with this change.

Reply:
Yes, I plan to squash it out of existence, sorry I should have made that more clear. I left it intact for a point of comparison whilst reviewing

Reply:
You could move it into a separate branch and raise another PR for it. It sounds like it might be useful behaviour to change, but I think I'd prefer to get some feedback from others first. Alternatively, we could just repeat the work another time..

Reply:
It certainly could form part of a rethink of what the `update` sub-command means, but it doesn't really make any sense in isolation. I'd rather not move it to its own draft PR as not to incur the wrath of the "long running PRs" metric.